solutions/bronze/cf-1624D.mdx

```diff
@@ -0,0 +1,120 @@
+---
+id: cf-1624D
+source: CF
+title: Palindromes Coloring
+author: Rameez Parwez
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/98942)
+
+## Explanation
+
+We can greedily solve this problem by first forming palindromic pairs from identical characters.
+We add one pair each of the `k` palindromes we want to build, as long as we have `k` pairs available.
+Once we are done adding pairs, we can still add one more character to the middle of each palindrome if there are atleast `k` individual characters remaining.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using std::cout;
+
+int main() {
+	int test_num;
+	std::cin >> test_num;
+	while (test_num--) {
+
+		int n, k;
+		std::string st;
+		std::array<int, 26> cnt{0};
+		std::cin >> n >> k >> st;
+
+		for (char x : st) { cnt[x - 'a']++; }
+
+		int pairs_cnt = 0, odd_cnt = 0;
```

Comment:
inb4 sans tells you to change it to `num_pairs` and `num_odd`

Reply:
real

Reply:
done